arroyo/processing/strategies/dead_letter_queue/policies/abstract.py

```diff
@@ -1,5 +1,53 @@
+import base64
 from abc import ABC, abstractmethod
-from typing import Any, Sequence
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Mapping, Optional, Sequence, Union
+
+Serializable = Union[str, bytes]
+
+
+@dataclass(frozen=True)
+class InvalidMessage:
+    """
+    A dataclass to generally represent a bad Kafka message.
+    The Kafka specific fields (offset etc) are all optional
+    since this can be used to pass almost anything to the
+    DLQ.
+
+    A `reason` can be provided for debugging purposes. If a
+    produce policy is configured on the relevant DLQ, a message
+    in the form returned by `to_dict()` will be produced via
+    the policy.
+
+    An `InvalidMessages` exception should be raised containing
+    one or more objects of this class in order to actually
+    pass data to the DLQ.
+    """
+
+    payload: Serializable
+    timestamp: datetime
+    reason: Optional[str] = None
+    consumer_group: Optional[str] = None
+    partition: Optional[int] = None
+    offset: Optional[int] = None
+
+    def to_dict(self) -> Mapping[str, str]:
+        if isinstance(self.payload, bytes):
+            try:
+                decoded = self.payload.decode("utf-8")
+            except UnicodeDecodeError:
+                decoded = "(base64) " + base64.b64encode(self.payload).decode("utf-8")
+        else:
+            decoded = self.payload
+        return {
+            "payload": decoded,
+            "timestamp": str(self.timestamp),
+            "reason": str(self.reason),
+            "consumer_group": str(self.consumer_group),
+            "partition": str(self.partition),
+            "offset": str(self.offset),
```

Comment:
Please do not turn None into strings. 
There is a difference between the String `None` and the JSON value for NULL. The first is a valid string, there will be no way to tell it apart from a valid string containing None and the actual NULL.
So please do not turn everything into strings. 

```
return {
    "payload": decoded,
    "timestamp": self.datetime,
    "reason": self.reason
    ....
```

In order to be able to json encode the message that contains a datetime please follow the pattern I mentioned here 
https://github.com/getsentry/arroyo/pull/57#discussion_r856621498

Reply:
https://github.com/getsentry/arroyo/pull/64